keanu-python/tests/test_base.py

```diff
@@ -0,0 +1,80 @@
+import keanu as kn
+import numpy as np
+import pytest
+from py4j.java_gateway import java_import
+
+jvm_view = kn.KeanuContext().jvm_view()
+java_import(jvm_view, "io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex")
+
+
+def test_can_pass_scalar_to_vertex():
+    gaussian = kn.Vertex(jvm_view.GaussianVertex, (0., 1.))
+    sample = gaussian.sample()
+
+    assert sample.isScalar()
+
+
+def test_can_pass_np_tensor_to_vertex():
+    gaussian = kn.Vertex(jvm_view.GaussianVertex, (np.array([[0.1, 0.4]]), np.array([[0.4, 0.5]])))
+    sample = gaussian.sample()
+
+    shape = sample.getShape()
+    assert sample.getRank() == 2
+    assert shape[0] == 1
+    assert shape[1] == 2
+
+
+def test_can_pass_java_object_to_vertex():
+    mu = kn.Vertex(jvm_view.GaussianVertex, (0., 1.))
+    gaussian = kn.Vertex(jvm_view.GaussianVertex, (mu, 1.))
+    sample = gaussian.sample()
+
+    assert sample.isScalar()
+
+
+def test_can_pass_array_to_vertex():
+    gaussian = kn.Vertex(jvm_view.GaussianVertex, ([3, 3], 0., 1.))
+    sample = gaussian.sample()
+
+    shape = sample.getShape()
+    assert sample.getRank() == 2
+    assert shape[0] == 3
+    assert shape[1] == 3
+
+
+def test_cannot_pass_generic_to_vertex():
+    class Temp:
```

Comment:
this needs a better name, like `GenericExampleClass`. If you call it `Temp` then it sounds like it should be deleted.